pkg/container-collection/tracerconnect/connect.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracerconnect
+
+import (
+	"fmt"
+	"os"
+
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type NetworkingTracer[Event any] interface {
+	Attach(key string, pid uint32, eventCallback func(Event)) error
+	Detach(key string) error
+}
+
+type ConnectNetworkingTracerToContainerCollectionArgs[Event any] struct {
+	Key           string
+	Tracer        NetworkingTracer[Event]
+	Resolver      containercollection.ContainerResolver
```

Comment:
IMHO, it is not really a "resolver" here, but more a "notifier" (about the creation/deletion of containers). The interface here should only need to define `Subscribe` (and `Unsubscribe`, if you decide to manage the closure internally, see [this comment](https://github.com/inspektor-gadget/inspektor-gadget/pull/1103/files#r1030312859)), all the Lookups are not actually needed (Maybe also `ConnectToContainerNotifier` would be a more generic name).

Maybe work for another PR?

Reply:
Yep, I think `Subscribe` and `Unsubscribe` can be moved out from that interface. Sure, let's do it in another PR.